package.json

```diff
@@ -42,10 +42,11 @@
     "@types/mocha": "^5.2.5",
     "@types/node": "^10.12.2",
     "bn.js": "^4.11.1",
-    "nyc": "^13.1.0",
+    "conditional-type-checks": "^0.5.0",
```

Comment:
This is a very low-star library (26 or so) and has a mid-download rate on npm often being < 100. For maintenance uncertainty as well as security reasons I wouldn't want to have this integrated as a standard dependency for the library.

Let's please stick to dependencies which are to some extend well-established in the ecosystem.

Reply:
Ah, just seeing that you are solving an issue with this. Hmm, any other way to achieve this? So I would get a bit more open to this then, but concerns remain.

Can we generally switch to have your ``typescript`` branch as a base branch and can you do further work on another branch as a PR towards this base branch? Then it get's easier to discuss further changes and things conflict less with the other work done in parallel.

Reply:
Yes, krzkaczor told me to do so on gitter. Sorry about that.
I think we can to it without this library. This was the "cleaner" way.

Reply:
No problem! 😄

If you haven't done much rebasing before, be careful, this is hairy. I am still in the process of always doing a local copy of the whole repo before every rebase, cause I am still always somewhat afraid.